bestie/server/test_result.go

```diff
@@ -57,6 +57,44 @@ func identifyStream(bazelBuildEvent bes.BuildEvent, streamId *build.StreamId) *b
 	return &stream
 }
 
+func readFileWithLimit(fileReader io.Reader) ([]byte, error) {
+	// Attempt to read the file contents all at once. If an
+	// EOF is encountered, then the file fit within the max
+	// size buffer, so it is eligible for processing. Otherwise,
+	// return an error indicating the file is too big.
+	// Adding 1 to the limit to know if the file size went over it.
+	var limit int = maxFileSize
+	buf := make([]byte, limit+1)
+	n, err := fileReader.Read(buf)
```

Comment:
I guess this is a bit more complicated than I figured - also forgot that io.Reader.Read can return without filling the whole buffer, so it has to be called in a loop. How about something that:

* creates a LimitedReader based on the io.Reader supplied
* does a ReadAll from the LimitedReader
* checks the underlying io.Reader to see if it has any more bytes, and fails if so

maybe like

```
limitRdr := io.LimitReader(fileReader, limit)
contents, err := io.ReadAll(limitRdr)
if err != nil {
  // failed to read
}

// check for one more byte
b := make([]byte, 1)
_, err := fileReader.Read(b)
if err == nil || !errors.Is(err, io.EOF) {
  // file too big, or some other error
}
// contents is valid + sufficiently small
```

This approach also avoids the buffer allocation of size limit

Limit should be passed in as a function arg with this approach

Reply:
I'll give this a try (probably not today though).